### DISCUSSION TITLE
Game Engine Architecture

### DISCUSSION THREAD

**Discussion by Hanggansta** [PanosK92/SpartanEngine#100] [General]
I have been following your game engine development. I noticed that in previous versions, you were using subsystems and a Context to manage all the subsystems. But now in the latest version, you changed all the subsystem functions to static. What is the reason for this change? What are the benefits of doing it this way? thank you very much.

**Comment by PanosK92** [comment]
The subsystems don't need multiple instances; therefore, you don't need to create objects out of them, therefore, they don't need to be anything but static classes. This way of thinking can be applied to many things in life, read more [here](https://en.wikipedia.org/wiki/Occam%27s_razor).
